src/pages/settings/AboutPage/ShareLogList/BaseShareLogList.tsx

```diff
@@ -1,66 +1,62 @@
-import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
-import {View} from 'react-native';
+import React, {useMemo, useRef} from 'react';
 import {withOnyx} from 'react-native-onyx';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
-import {usePersonalDetails} from '@components/OnyxProvider';
-import OptionsSelector from '@components/OptionsSelector';
+import {useBetas, usePersonalDetails} from '@components/OnyxProvider';
 import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useDebouncedState from '@hooks/useDebouncedState';
 import useLocalize from '@hooks/useLocalize';
 import useNetwork from '@hooks/useNetwork';
-import useThemeStyles from '@hooks/useThemeStyles';
 import * as FileUtils from '@libs/fileDownload/FileUtils';
 import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import type {Report} from '@src/types/onyx';
 import type {BaseShareLogListOnyxProps, BaseShareLogListProps} from './types';
 
-function BaseShareLogList({betas, reports, onAttachLogToReport}: BaseShareLogListProps) {
-    const [searchValue, setSearchValue] = useState('');
-    const [searchOptions, setSearchOptions] = useState<Pick<OptionsListUtils.GetOptions, 'recentReports' | 'personalDetails' | 'userToInvite'>>({
-        recentReports: [],
-        personalDetails: [],
-        userToInvite: null,
-    });
-
+function BaseShareLogList({reports, onAttachLogToReport}: BaseShareLogListProps) {
+    const [searchValue, debouncedSearchValue, setSearchValue] = useDebouncedState('');
     const {isOffline} = useNetwork();
     const {translate} = useLocalize();
-    const styles = useThemeStyles();
     const isMounted = useRef(false);
     const personalDetails = usePersonalDetails();
+    const betas = useBetas();
+
+    const searchOptions = useMemo(() => {
+        const isOptionsDataReady = ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(personalDetails);
 
-    const updateOptions = useCallback(() => {
+        if (!isMounted.current || !isOptionsDataReady) {
+            isMounted.current = true;
```

Comment:
`isMounted` ref is strange to me as well. Is it really necessary? How could the options data be ready by the component not be mounted?

It seems like it would be better to:
- use a `ref` for `searchOptions` and have it's default value be what is in this early return
- Update the `searchOptions` ref inside this `useMemo()`

Reply:
Alternative thought: Maybe each of these 4 properties should be their own distinct variable. Would that clean up the code at all?

Reply:
yeah, I kinda bluntly repurposed `isMounted` flag from the previous version of the component, but I agree, with `isOptionsDataReady` it's totally unnecessary.